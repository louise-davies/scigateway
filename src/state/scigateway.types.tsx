import { Step } from 'react-joyride';
import { Theme } from '@material-ui/core/styles/createMuiTheme';

export const AuthSuccessType = 'scigateway:auth_success';
export const AuthFailureType = 'scigateway:auth_failure';
export const LoadingAuthType = 'scigateway:loading_auth';
export const LoadedAuthType = 'scigateway:loaded_auth';
export const LoadAuthProviderType = 'scigateway:loading_auth_provider';
export const LoadScheduledMaintenanceStateType =
  'scigateway:load_scheduled_maintenance_state';
export const LoadMaintenanceStateType = 'scigateway:load_maintenance_state';
export const InvalidateTokenType = 'scigateway:api:invalidate_token';
export const ConfigureStringsType = 'scigateway:configure_strings';
export const LoginType = 'scigateway:login';
export const NotificationType = 'scigateway:api:notification';
export const RegisterRouteType = 'scigateway:api:register_route';
export const RequestPluginRerenderType = 'scigateway:api:plugin_rerender';
export const SendThemeOptionsType = 'scigateway:api:send_themeoptions';
export const LoadDarkModePreferenceType =
  'scigateway:load_dark_mode_preference';
export const SignOutType = 'scigateway:signout';
export const ToggleDrawerType = 'scigateway:toggledrawer';
export const DismissNotificationType = 'scigateway:dismissnotification';
export const ConfigureFeatureSwitchesType = 'scigateway:feature_switches';
export const AddNotificationsType = 'scigateway:add_notification';
export const SiteLoadingType = 'scigateway:site_loading';
export const ConfigureAnalyticsType = 'scigateway:configure_analytics';
export const InitialiseAnalyticsType = 'scigateway:initialise_analytics';
export const ToggleHelpType = 'scigateway:toggle_help';
export const AddHelpTourStepsType = 'scigateway:add_help_tour_steps';
export const RegisterStartUrlType = 'scigateway:register_start_url';

export interface NotificationPayload {
  message: string;
  severity: string;
}

export interface AddNotificationsPayload {
  notifications: NotificationPayload[];
}

export interface AppStrings {
  [id: string]: string;
}

export interface ConfigureStringsPayload {
  res: ApplicationStrings;
}

export interface ApplicationStrings {
  [section: string]: AppStrings;
}

export interface FeatureSwitchesPayload {
  switches: FeatureSwitches;
}

export interface FeatureSwitches {
  showContactButton: boolean;
  showHelpPageButton: boolean;
}

export interface StartUrlPayload {
  startUrl: string;
}

export interface RegisterRoutePayload {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  order: number;
  helpText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface PluginConfig {
  section: string;
  link: string;
  plugin: string;
  displayName: string;
  order: number;
  helpText?: string;
  helpSteps?: { target: string; content: string }[];
}

export interface GroupedPlugins {
  [section: string]: PluginConfig[];
}

export interface LoginPayload {
  username: string;
  password: string;
}

export interface AccessPayload {
  authorisation: boolean;
  accessToken: string;
  error: null;
}

export interface AuthorisedPayload {
  token: string;
}

export interface SendThemeOptionsPayload {
  theme: Theme;
  broadcast: boolean;
}

export interface LoadDarkModePreferencePayload {
  darkMode: boolean;
}

export interface DismissNotificationPayload {
  index: number;
}

export interface AuthProviderPayload {
  authProvider: string;
  authUrl: string | undefined;
}

export interface ScheduledMaintenanceStatePayLoad {
  scheduledMaintenance: ScheduledMaintenanceState;
}

export interface ScheduledMaintenanceState {
  show: boolean;
  message: string;
}

export interface MaintenanceStatePayLoad {
  maintenance: MaintenanceState;
}

export interface MaintenanceState {
  show: boolean;
  message: string;
}

export interface SiteLoadingPayload {
  loading: boolean;
}

export interface ConfigureAnalyticsPayload {
  id: string;
}

export interface AddHelpTourStepsPayload {
  steps: Step[];
}
