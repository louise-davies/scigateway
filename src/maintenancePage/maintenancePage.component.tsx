import React from 'react';
import Typography from '@material-ui/core/Typography';
import {
  Theme,
  StyleRules,
  createStyles,
  WithStyles,
  withStyles,
} from '@material-ui/core';
import { connect } from 'react-redux';
import { StateType } from '../state/state.types';

const styles = (theme: Theme): StyleRules =>
  createStyles({
    root: {
      padding: theme.spacing(2),
      backgroundColor: theme.palette.background.default,
      textAlign: 'center',
    },
    container: {
      display: 'flex',
      flexDirection: 'column',
      marginTop: theme.spacing(2),
      marginBottom: theme.spacing(2),
      color: theme.palette.text.primary,
    },
    titleText: {
      fontWeight: 'bold',
      color: theme.palette.secondary.main,
    },
  });

interface MaintenancePageProps {
  message: string;
}

export type CombinedMaintenancePageProps = MaintenancePageProps &
  WithStyles<typeof styles>;

const MaintenancePage = (
  props: CombinedMaintenancePageProps
): React.ReactElement => {
  return (
    <div className={props.classes.root}>
      <Typography variant="h3" className={props.classes.titleText}>
        Maintenance
      </Typography>
      <div className={props.classes.container}>
        <Typography variant="h4">{props.message}</Typography>
      </div>
    </div>
  );
};

const mapStateToProps = (state: StateType): MaintenancePageProps => ({
  message: state.scigateway.maintenance.message,
});

const MaintenancePageWithStyles = withStyles(styles)(MaintenancePage);

export default connect(mapStateToProps)(MaintenancePageWithStyles);
