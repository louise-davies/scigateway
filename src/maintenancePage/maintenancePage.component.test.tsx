import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { default as MaintenancePage } from './maintenancePage.component';
import { StateType } from '../state/state.types';
import { authState, initialState } from '../state/reducers/scigateway.reducer';

describe('Maintenance page component', () => {
  let mockStore;
  let state: StateType;

  beforeEach(() => {
    mockStore = configureStore();
    state = {
      scigateway: { ...initialState, authorisation: { ...authState } },
    };
    state.scigateway.maintenance.message = 'test';
  });

  it('should render correctly', () => {
    const testStore = mockStore(state);
    const wrapper = shallow(<MaintenancePage store={testStore} />)
      .dive()
      .dive();
    expect(wrapper).toMatchSnapshot();
  });
});
