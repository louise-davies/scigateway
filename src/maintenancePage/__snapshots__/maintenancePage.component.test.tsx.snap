// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maintenance page component should render correctly 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "backgroundColor": "background.default",
      "padding": 2,
      "textAlign": "center",
    }
  }
>
  <ForwardRef(Typography)
    sx={
      Object {
        "color": "secondary.main",
        "fontWeight": "bold",
      }
    }
    variant="h3"
  >
    Maintenance
  </ForwardRef(Typography)>
  <ForwardRef(Box)
    sx={
      Object {
        "color": "text.primary",
        "display": "flex",
        "flexDirection": "column",
        "marginBottom": 2,
        "marginTop": 2,
      }
    }
  >
    <ForwardRef(Typography)
      variant="h4"
    >
      test
    </ForwardRef(Typography)>
  </ForwardRef(Box)>
</ForwardRef(Box)>
`;
